src/vms/pvm/etna-builder/spendHelper.ts

```diff
@@ -0,0 +1,255 @@
+import type { TransferableOutput } from '../../../serializable';
+import { TransferableInput } from '../../../serializable';
+import type { Utxo } from '../../../serializable/avax/utxo';
+import { bigIntMin } from '../../../utils/bigintMath';
+import { compareTransferableOutputs } from '../../../utils/sort';
+import type { Dimensions } from '../../common/fees/dimensions';
+import { addDimensions, dimensionsToGas } from '../../common/fees/dimensions';
+import { getInputComplexity, getOutputComplexity } from '../txs/fee';
+
+interface SpendHelperProps {
+  changeOutputs: readonly TransferableOutput[];
+  complexity: Dimensions;
+  gasPrice: bigint;
+  inputs: readonly TransferableInput[];
+  stakeOutputs: readonly TransferableOutput[];
+  toBurn: Map<string, bigint>;
+  toStake: Map<string, bigint>;
+  weights: Dimensions;
+}
+
+/**
+ * The SpendHelper class assists in managing and processing the spending of assets,
+ * including handling complexities, gas prices, and various outputs and inputs.
+ *
+ * @class
+ */
+export class SpendHelper {
+  private readonly gasPrice: bigint;
+  private readonly toBurn: Map<string, bigint>;
+  private readonly toStake: Map<string, bigint>;
+  private readonly weights: Dimensions;
+
+  private complexity: Dimensions;
+  private changeOutputs: readonly TransferableOutput[];
+  private inputs: readonly TransferableInput[];
+  private stakeOutputs: readonly TransferableOutput[];
+
+  private inputUTXOs: readonly Utxo[] = [];
+
+  constructor({
+    changeOutputs,
```

Comment:
No I don't think so since we are doing a single object parameter.

I know it's a bit more verbose, but I personally prefer it over having 3+ arguments. Named props read better imo.